### PR TITLE
[FrameworkBundle] Allow to avoid `limit` definition in a RateLimiter configuration when using the `no_limit` policy

### LINKED ISSUE #48268: [FrameworkBundle][RateLimiter] `no_limit` policy requires a limit to be configured
### Symfony version(s) affected

From v5.2.0-BETA3

### Description

#38665 exposed the `no_limit` policy to the configuration without taking its singularity into account. As such it requires the lock component to be installed and enabled, but also a `limit` to be configured (it would be ignored).

### How to reproduce

Configure a limiter with the `no_limit` policy:

```yaml
framework:
    rate_limiter:
        no_limit:
            policy: no_limit
```
and you will get an `InvalidConfigurationException`:

> The child config "limit" under "framework.rate_limiter.limiters.no_limit" must be configured: The maximum allowed hits in a fixed interval or burst

### Possible Solution

Do not mark `limit` as required and validate it’s set only for policies other than `no_limit`.

### Additional Context

I don’t think needing to enable the lock component to configure a `no_limit` limiter really is an issue, because you would not use only this policy.

### PR DESCRIPTION
**Pull request by alexandre-daubois**
| Q             | A
| ------------- | ---
| Branch?       | 6.3
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no
| Tickets       | Fix #48268
| License       | MIT
| Doc PR        | N/A

As stated in the issue, DX will be improved if we can avoid setting a `limit` when using `no_limit` policy.
Targets 6.3, as it can be considered as a feature.

### CODE REVIEW

**Comment by fabpot** [comment]
This PR should target 6.3.

**Comment by alexandre-daubois** [comment]
Updated base branch + added the related CHANGELOG.md entry.

**Comment by MatTheCat** [comment]
This seems like a bugfix to me :thinking:

**Comment by alexandre-daubois** [comment]
@MatTheCat I understand your point of view and went for a bugfix as well at first sight. Fundamentally, the `no_limit` policy works well without this PR. Here, we actually "just" improve the developer experience. I think this is why it is more considered as a feature, rather than a bugfix.

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -2119,6 +2118,10 @@ private function addRateLimiterSection(ArrayNodeDefinition $rootNode, callable $
                                         ->end()
                                     ->end()
                                 ->end()
+                                ->validate()
+                                    ->ifTrue(function ($v) { return 'no_limit' !== $v['policy'] && null === ($v['limit'] ?? null); })
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php]: `!isset()` is just fine, let's not use fancy syntax when not needed.

**Review by nicolas-grekas — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @alexandre-daubois.

**Pull request merged**